keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -32,56 +134,155 @@
+and make progress.
+-->
+
+None.
 
 ## Proposal
 
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
 ### User Stories
 
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
 #### Story 1
-As a User of Kubernetes, I should be able to update my StatefulSet, more than one Pod at a time, in a 
-RollingUpdate manner, if my Stateful app can tolerate more than one pod being down, thus allowing my 
-update to finish much faster. 
+
+As a User of Kubernetes, I should be able to update my StatefulSet, more than one Pod at a time, in a
+RollingUpdate manner, if my Stateful app can tolerate more than one pod being down, thus allowing my
+update to finish much faster.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+With `maxUnavailable` feature enabled, we'll bring down more than one pod at a time, if your application can't tolerate
+this behavior, you should absolutely leave the field unset or use 1, which will match default behavior.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+We are proposing a new field called `maxUnavailable` whose default value will be 1. In this mode, StatefulSet will behave exactly like its current behavior.
+Its possible we introduce a bug in the implementation. The mitigation currently is that is disabled by default in Alpha phase for people to try out and give
+feedback.
+In Beta phase when its enabled by default, people will only see issues or bugs when `maxUnavailable` is set to something greater than 1. Since people have
+tried this feature in Alpha, we would have time to fix issues.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
 
 ### Implementation Details
```

Comment:
Nit: while editing this document, can you drop the json and protobuf tags from the API below leaving ONLY the type information.